lib/parser/lexer.rl

```diff
@@ -1746,6 +1746,15 @@ class Parser::Lexer
       # SYMBOL LITERALS
       #
 
+      # :&&, :||
+      ':' % { emit(:tSYMBEG) }
+      ( '&&' % { op_tok = :tANDOP }
+      | '||' % { op_tok = :tOROP  }
+      )
+      => {
+        emit(op_tok, tok(@te - 2, @te), @te - 2, @te)
```

Comment:
We typically do it like the following elsewhere:
```
# :&&, :||
':' ("&&" | "||") => {
  fhold; fhold; emit(:tSYMEG, @ts, @ts + 1)
}
```

Reply:
But then the lexer emits `tAMPER` (because there's no transition to the state that can handle `&&` as a `tANDOP`).  Adding `fgoto expr_end` (like in `1 && 2`) solves it, but is it a good idea?

Reply:
Now that I look at MRI, why not switch to EXPR_FNAME after `:` that's not `:'` or `:"` ?